### PR TITLE
Add regression test for GEN_ADD's REAL left-fold overflow-to-INF behavior

### PR DESCRIPTION
**Pull request by franz-ms-muc**
IEC 61131-3 specifies ADD_n as the equivalent of IN1 + IN2 + ... + INn, evaluated as a left fold: ((IN1 + IN2) + ...) + INn. For REAL/LREAL, an intermediate step overflowing to IEEE-754 +/-INF stays that way for the rest of the fold, even when the true mathematical result of the whole expression is a perfectly valid, finite, in-range value -- floating-point arithmetic is simply not associative.

ADD_3(3.0e38, 3.0e38, -3.0e38): true result is 3.0e38 (well within REAL's ~3.4e38 max), but (IN1+IN2) overflows to +INF first, and +INF + IN3 stays +INF.

This is expected, standard-conformant behavior, not a defect -- see the discussion on #1029, which was closed as "no real issue in the code," with follow-up work moving to documentation instead. No existing test covered ADD_n with more than 2 inputs at all, so this locks the behavior in as a regression test rather than tracking it as something to fix.

- #1029

Co-Authored-By: Claude Sonnet 5 <noreply@anthropic.com>

### CODE REVIEW

File: tests/modules/IEC61131-3/GEN_ADD_tester.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+using namespace forte::literals;
+
+namespace forte::iec61131::arithmetic {
+  // See https://github.com/eclipse-4diac/4diac-forte/issues/1029: the variadic ADD_n FB folds
+  // its inputs pairwise, left to right. For REAL/LREAL this means an intermediate step that
+  // overflows to IEEE-754 +/-INF can never recover, even when the true mathematical result of
+  // the whole expression is a perfectly valid, finite, in-range value -- unlike integer types,
+  // where two's-complement wraparound is associative and always yields the correct wrapped total
+  // regardless of intermediate grouping.
```
[the remark this answers is not in the document]
Comment (mx990) [reply tests/modules/IEC61131-3/GEN_ADD_tester.cpp]: I agree with Copilot here, IEC 61131-3 also strictly considers signed integer overflow an error.

File: tests/modules/IEC61131-3/GEN_ADD_tester.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  };
+
+  BOOST_FIXTURE_TEST_SUITE(GenAdd3RealOverflowTests, GenAdd3RealOverflowFixture)
+
+  // IN1 + IN2 + IN3 == 3.0e38, a normal finite REAL value (REAL max is ~3.4e38). But the fold
+  // evaluates (IN1 + IN2) first, which overflows to +INF, and +INF + IN3 stays +INF -- so the
+  // reported OUT loses all information about the true, well-defined, in-range total. This test
+  // documents that known limitation (tracked in the issue above); it is expected to start
+  // failing once the fold is fixed to no longer be order-of-evaluation dependent, at which point
+  // it should be updated to assert the correct finite result instead.
```
Comment (mx990) [inline tests/modules/IEC61131-3/GEN_ADD_tester.cpp]: This comment assumes that the current behavior is wrong somehow. However, as far as I can tell, the current behavior is perfectly valid and well-defined according to both IEEE 754 and IEC 61131-3. I therefore see no need to change or *fix* anything here.

**Review by mx990 — CHANGES_REQUESTED** [review]

File: tests/modules/IEC61131-3/GEN_ADD_tester.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  BOOST_AUTO_TEST_CASE(intermediateOverflowToInfCorruptsInRangeResult) {
+    mIn1 = 3.0e38_REAL;
+    mIn2 = 3.0e38_REAL;
+    mIn3 = -3.0e38_REAL;
+    triggerEvent(REQ);
+    BOOST_CHECK(checkForSingleOutputEventOccurence(CNF));
+    BOOST_CHECK(std::isinf(static_cast<CIEC_REAL::TValueType>(mOut)));
```
[the remark this answers is not in the document]
Comment (franz-ms-muc) [reply tests/modules/IEC61131-3/GEN_ADD_tester.cpp]: Good catch. Now checks the sign too (value > 0), not just std::isinf(), since the expected result here is specifically +INF. See e3c4d9b29.

File: tests/modules/IEC61131-3/GEN_ADD_tester.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (franz-ms-muc) [reply tests/modules/IEC61131-3/GEN_ADD_tester.cpp]: Correct — signed overflow is UB in C++, not guaranteed wraparound. Narrowed the claim to unsigned integer types and called out that signed overflow is UB by contrast. See b119eafdc.

[the remark this answers is not in the document]
Comment (franz-ms-muc) [reply tests/modules/IEC61131-3/GEN_ADD_tester.cpp]: Good point, added — the comment now also notes IEC 61131-3 treats signed overflow as an error condition itself, not just a C++ implementation-level UB concern. See ad01f1e1b.

File: tests/modules/IEC61131-3/GEN_ADD_tester.cpp (hunk 2 above)
Comment (franz-ms-muc) [reply tests/modules/IEC61131-3/GEN_ADD_tester.cpp]: Agreed — consistent with the resolution on #1029 (closed as no real code issue, IEC 61131-3's left-fold ADD_n definition + IEEE 754 non-associativity explain it fully). Reframed the test: it no longer calls this a limitation to be fixed, but documents/locks in the correct, specified left-fold behavior as a regression test (renamed the test case accordingly). PR title/description and commit message updated to match. See 60aaeb117.

**Review by mx990 — APPROVED** [review]
